src/Azure/Shared/CosmosDB/AzureCosmosDBOptions.cs

```diff
@@ -0,0 +1,116 @@
+    /// </summary>
+    public string Database { get; set; } = ORLEANS_DB;
+
+    /// <summary>
+    /// The name of the container to use to store clustering information.
+    /// </summary>
+    public string Container { get; set; } = default!;
+
+    /// <summary>
+    /// RU units for container, can be set to 0 if throughput is specified on database level. See https://docs.microsoft.com/en-us/azure/cosmos-db/set-throughput
+    /// </summary>
+    public int ContainerThroughput { get; set; } = ORLEANS_DEFAULT_RU_THROUGHPUT;
+
+    /// <summary>
+    /// The connection mode to use when connecting to the azure cosmos DB service.
+    /// </summary>
+    [JsonConverter(typeof(JsonStringEnumConverter))]
+    public ConnectionMode ConnectionMode { get; set; } = ConnectionMode.Direct;
+
+    /// <summary>
+    /// Delete the database on initialization.  Useful for testing scenarios.
+    /// </summary>
+    public bool DropDatabaseOnInit { get; set; }
+
+    /// <summary>
+    /// The throughput mode to use for the container used for clustering.
+    /// </summary>
+    /// <remarks>If the throughput mode is set to Autoscale then the <see cref="ContainerThroughput"/> will need to be at least 4000 RUs.</remarks>
+    public ThroughputMode ThroughputMode { get; set; } = ThroughputMode.Manual;
+
+    /// <summary>
+    /// The Azure Active Directory credentials.
+    /// </summary>
+    /// <remarks>If this is set then the <see cref="AccountKey"/> will be ignored.</remarks>
+    public TokenCredential? Credential { get; set; }
+
+    /// <summary>
+    /// Json Serializer options.
+    /// </summary>
+    public JsonSerializerOptions? SerializerOptions { get; set; }
+
+    /// <summary>
+    /// Factory method for creating a <see cref="CosmosClient"/>.
+    /// </summary>
+    /// <remarks>If this method is specified, the <see cref="AccountEndpoint"/> and <see cref="AccountKey"/> will be ignored.</remarks>
+    public Func<IServiceProvider, ValueTask<CosmosClient>>? CosmosDBClientFactory { get; set; }
+
+    // TODO: Consistency level for emulator (defaults to Session; https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator)
+    internal IndexingMode? GetConsistencyLevel() => !string.IsNullOrWhiteSpace(this.AccountEndpoint) && this.AccountEndpoint.Contains("localhost") ? IndexingMode.None : null;
+
+    internal ThroughputProperties? GetThroughputProperties() =>
+        this.ThroughputMode switch
+        {
+            ThroughputMode.Manual => ThroughputProperties.CreateManualThroughput(this.ContainerThroughput),
+            ThroughputMode.Autoscale => ThroughputProperties.CreateAutoscaleThroughput(
+                this.ContainerThroughput == 400 ? 4000 : this.ContainerThroughput),
+            ThroughputMode.Serverless => null,
+            _ => throw new ArgumentOutOfRangeException(nameof(this.ThroughputMode), $"There is no setup for throughput mode {this.ThroughputMode}")
+        };
+}
+
+public enum ThroughputMode
```

Comment:
Maybe worth moving this to a separate file?

Reply:
Done